jupyterlab_git/tests/test_branch.py

```diff
@@ -1,874 +1,831 @@
+                    "",
+                    "fatal: No tags can describe '01234567899999abcdefghijklmnopqrstuvwxyz'.",
+                )
+            ),
+        ]
+
+        # When
+        with pytest.raises(Exception) as error:
+            await Git(FakeContentManager("/bin"))._get_tag(
+                current_path="test_curr_path", commit_sha="blah"
+            )
+
+        assert (
+            "Error [fatal: Not a valid object name blah] "
+            "occurred while executing [git describe --tags blah] command to get nearest tag associated with branch."
+            == str(error.value)
+        )
+
+        actual_response = await Git(FakeContentManager("/bin"))._get_tag(
+            current_path="test_curr_path",
+            commit_sha="01234567899999abcdefghijklmnopqrstuvwxyz",
+        )
+
+        assert actual_response is None
+
+        # Then
+        mock_execute.assert_has_calls(
+            [
+                call(
+                    ["git", "describe", "--tags", "blah"],
+                    cwd=os.path.join("/bin", "test_curr_path"),
+                ),
+                call(
+                    [
+                        "git",
+                        "describe",
+                        "--tags",
+                        "01234567899999abcdefghijklmnopqrstuvwxyz",
+                    ],
+                    cwd=os.path.join("/bin", "test_curr_path"),
+                ),
+            ],
+            any_order=False,
+        )
+
+
+@pytest.mark.asyncio
+async def test_no_tags():
+    with patch("jupyterlab_git.git.execute") as mock_execute:
+        # Given
+        mock_execute.return_value = tornado.gen.maybe_future(
+            (128, "", "fatal: No names found, cannot describe anything.\n")
+        )
+
+        # When
+        actual_response = await Git(FakeContentManager("/bin"))._get_tag(
+            "/path/foo", "768c79ad661598889f29bdf8916f4cc488f5062a"
+        )
+
+        # Then
+        mock_execute.assert_called_once_with(
+            ["git", "describe", "--tags", "768c79ad661598889f29bdf8916f4cc488f5062a"],
+            cwd="/path/foo",
+        )
+        assert actual_response is None
+
+
+@pytest.mark.asyncio
+async def test_branch_success():
+    with patch("jupyterlab_git.git.execute") as mock_execute:
+        # Given
+        process_output_heads = [
+            "feature-foo\tabcdefghijklmnopqrstuvwxyz01234567890123\torigin/feature-foo\t*",
+            "master\tabcdefghijklmnopqrstuvwxyz01234567890123\torigin/master\t ",
+            "feature-bar\t01234567899999abcdefghijklmnopqrstuvwxyz\t\t ",
+        ]
+        process_output_remotes = [
+            "origin/feature-foo\tabcdefghijklmnopqrstuvwxyz01234567890123",
+            "origin/master\tabcdefghijklmnopqrstuvwxyz01234567890123",
+        ]
+
+        mock_execute.side_effect = [
+            # Response for get all refs/heads
+            tornado.gen.maybe_future((128, "\n".join(process_output_heads), "")),
```

Comment:
The test is failing because the return code here needs to be 0, since this is meant to represent a successful response

Reply:
Thanks a lot @telamonian and sorry I should have seen that one...